### PR TITLE
Made most libraries panic on js targets instead of not compiling

### PR DESCRIPTION
**Pull request by DragosPopse**
In reference to #2388 . Writing cross platform code for wasm + desktop gets quite annoying when you can't enclose your library usage in `when` statements. This PR fixes the issue by implementing all non supported procedures as a simple `panic` call. This will allow usage of those libraries inside `when ODIN_OS == ...` statements, and also give better error messages when you are using a procedure that is not supported on your current platform.

### CODE REVIEW

File: core/dynlib/lib_js.odin [hunk 1]
```diff
@@ -0,0 +1,15 @@
...
+	panic("core:dynlib not supported by JS target")
+}
+
+_unload_library :: proc(library: Library) -> bool {
+	panic("core:dynlib not supported by JS target")
+}
+
+_symbol_address :: proc(library: Library, symbol: string) -> (ptr: rawptr, found: bool) {
+	panic("core:dynlib not supported by JS target")
+}
```
Comment (gingerBill) [inline core/dynlib/lib_js.odin:15]: Would this be better to just return `false` for all of these?

Comment (Kelimion) [reply core/dynlib/lib_js.odin:15]: Maybe, but if not, then replace `panic("...")` with `unimplemented("...")`.

Comment (DragosPopse) [reply core/dynlib/lib_js.odin:15]: I didn't know `unimplemented` exists. What is the difference between that and `panic`? But yeah maybe for the dynlib stuff a false could be nicer to use than straight busting your program. I'll update it

**Comment by DragosPopse** [comment]
Also, coming back to my discord post. Would it be sensible to apply the same treatment to `intrinsics.wasm_grow_memory` ? I'm asking because I am not particularly sure how would I do that in a builtin proc.

File: core/dynlib/lib_js.odin (hunk 1 above)
Comment (Kelimion) [reply core/dynlib/lib_js.odin]: It's still a `panic`, only it makes it easier to search for things that are `unimplemented`.

Comment (DragosPopse) [reply core/dynlib/lib_js.odin]: Good call, i'll update my pr with your suggestion in a few hours

**Comment by DragosPopse** [comment]
Updated. Let me know if anything else is needed

**Comment by Kelimion** [comment]
> Also, coming back to my discord post. Would it be sensible to apply the same treatment to `intrinsics.wasm_grow_memory` ? I'm asking because I am not particularly sure how would I do that in a builtin proc.

That's a @gingerBill question.

**Comment by gingerBill** [comment]
No because it is a built-in intrinsic and something at the wasm level itself.

**Pull request merged**